Neos.Neos/Classes/Utility/NodeUriPathSegmentGenerator.php

```diff
@@ -38,19 +41,36 @@ class NodeUriPathSegmentGenerator
      */
     public function generateUriPathSegment(?Node $node = null, ?string $text = null): string
     {
-        $language = null;
-        if ($node) {
-            $text = $text ?: $node->getLabel() ?: ($node->nodeName?->value ?? '');
-            $languageDimensionValue = $node->originDimensionSpacePoint->coordinates['language'] ?? null;
-            if (!is_null($languageDimensionValue)) {
-                $locale = new Locale($languageDimensionValue);
-                $language = $locale->getLanguage();
-            }
-        } elseif (is_null($text) || empty($text)) {
+        if ($node === null && empty($text)) {
             throw new Exception('Given text was empty.', 1457591815);
         }
-        $text = $this->transliterationService->transliterate($text, $language);
 
-        return Transliterator::urlize($text);
+        $textForNode = $text ?: $node->getLabel() ?: $node->nodeName?->value ?? '';
+
+        return $this->generateUriPathSegmentFromTextForDimension(
+            $textForNode,
+            $node?->originDimensionSpacePoint->toDimensionSpacePoint() ?? DimensionSpacePoint::fromArray([])
+        );
+    }
+
+    /**
+     * Generates a URI path segment for a given text taking the language dimension value into account
+     *
+     * @param string $text
+     * @param DimensionSpacePoint $dimensionSpacePoint to determine language dimension value from
+     */
+    public function generateUriPathSegmentFromTextForDimension(string $text, DimensionSpacePoint $dimensionSpacePoint): string
```

Comment:
`generateUriPathSegmentForLocale`


and new locale service to `createLocaleFromDimensionSpacePointAndContentRepositoryId`